src/DynamoCore/Extensions/LinterExtensionBase.cs

```diff
@@ -157,9 +181,14 @@ private void InitializeRules()
         ///<inheritdoc/>
         public virtual void Ready(ReadyParams sp)
         {
+            currentWorkspace = null;
+            HomeWorkspaceModel newWorkspace = sp.CurrentWorkspaceModel as HomeWorkspaceModel;
+            if (newWorkspace != null && !String.IsNullOrEmpty(newWorkspace.FileName))
+            {
+                currentWorkspace = newWorkspace;
+            }
+
             ReadyParamsRef = sp;
-            if (IsActive)
```

Comment:
nothing meaningful was ever happening here, this is too early, the workspace is not filled yet.